### DISCUSSION TITLE
Provide server response for ILP over TCP

### DISCUSSION THREAD

**Discussion by spxie85** [questdb/questdb#1184] [Ideas]
QuestDB does not currently provide a server response for ILP messages sent over TCP, which means the sender would not know of any errors QuestDB encounters e.g. INVALID_FIELD_VALUE, rows out of order etc., nor will it know if the messages were successfully ingested.

**Comment by bluestreak01** [comment]
Providing responses will generally slow down ILP and make interaction more complicated, e.g. you'd have to read fragmented TCP packets from socket.

This generally means that driver will be required.

Influx do this over HTTP, which is another envelope to parse. I wonder if there is another way to troubleshoot errors in ILP ? Perhaps dump failed lines to a text file somewhere ?

**Reply by spxie85** [reply]
That was the intention, if there's a way for the sending application to know which ILP has failed.

**Reply by bluestreak01** [reply]
Would you like to know on programmatic level that ILP line didn't go thru? Or an alert or notification would suffice? I'm trying to avoid sending "ok" for each successful line. This would cause massive overhead on both ends.

**Reply by spxie85** [reply]
Is it feasible to have a toggle to enabled ingestion status reporting so one can debug more easily which ILP record is causing issues? To prevent a performance impact this property could be disabled by default e.g. when running in production.
